src/Blazored.Modal/BlazoredModalInstance.razor.cs

```diff
@@ -95,20 +96,28 @@ public async Task CloseAsync()
     public async Task CloseAsync(ModalResult modalResult)
     {
         // Fade out the modal, and after that actually remove it
-        if (AnimationType is ModalAnimationType.FadeInOut)
+        if (AnimationType is not ModalAnimationType.None && AnimationType is not ModalAnimationType.FadeIn && 
+            AnimationType is not ModalAnimationType.MoveIn && AnimationType is not ModalAnimationType.PopIn)
         {
-            OverlayAnimationClass += " fade-out";
+            OverlayAnimationClass = AnimationType switch
+            {
+                ModalAnimationType.FadeInOut or ModalAnimationType.FadeOut or
+                ModalAnimationType.MoveInOut or ModalAnimationType.MoveOut or
+                ModalAnimationType.PopInOut => "blazored-fade-out",
+                _ => string.Empty,
+            };
+            
+            ModalAnimationClass = AnimationType switch
+            {
+                ModalAnimationType.FadeInOut or ModalAnimationType.FadeOut => "blazored-fade-out",
+                ModalAnimationType.MoveInOut or ModalAnimationType.MoveOut => "blazored-move-out",
+                ModalAnimationType.PopInOut => "blazored-pop-out",
```

Comment:
can we revert the CSS class name changes, I can't imagine there is any value in changing them and it would make this a breaking change which I'd like to avoid

Reply:
Fixed